cartridges/int_algolia_sfra/cartridge/controllers/Cart.js

```diff
@@ -24,4 +26,45 @@ server.append('Show', function (req, res, next) {
     next();
 });
 
+
+server.append('AddProduct', function (req, res, next) {
+    if (algoliaData.getPreference('Enable') && algoliaData.getPreference('EnableInsights')) {
+
+        var productId = req.form.pid;
+
+        var algoliaProductData = {};
+        algoliaProductData.pid = productId;
+
+        if (algoliaData.getPreference('RecordModel') === MASTER_LEVEL) {
+            var product = ProductMgr.getProduct(productId);
+
+            if (product && (product.isMaster() || product.isVariationGroup())) {
+                algoliaProductData.pid = product.ID;
```

Comment:
You forgot the `else` clause to set `algoliaProductData.pid = product.masterProduct.ID`

Reply:
on line 36, we covered it

Reply:
In that case the `pid` is the variantID isn't it?

Reply:
You had initially fixed it: https://github.com/algolia/algoliasearch-sfcc-b2c/pull/178/commits/b9ac9665fc8673ceeedcaa2305ade1426abc9a13#diff-48b65d55ceb98f4b440f39608b808768eff7ffffad1110230cbb0c5590ac2aedR39